buildscripts/common.props

```diff
@@ -1,21 +1,35 @@
 <Project>
-  <PropertyGroup>
-    <VersionPrefix>5.0.0</VersionPrefix>
-    <NoWarn>$(NoWarn);CS1591</NoWarn>
-    <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
-  </PropertyGroup>
+	<PropertyGroup>
+		<NoWarn>$(NoWarn);CS1591;CS3014;CS3003;CS3001;CS3021</NoWarn>
+		<RepositoryType>git</RepositoryType>
+		<RepositoryUrl>https://github.com/castleproject/Core</RepositoryUrl>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'==''">4.0.0</BuildVersion>
+		<BuildVersion Condition="'$(APPVEYOR_BUILD_VERSION)'!=''">$(APPVEYOR_BUILD_VERSION)</BuildVersion>
```

Comment:
I'd prefer we used something more generic, than had the top level script populate BUILD_VERSION or something so we can do the same thing on Travis.

Not sure if it breaks anything with the new tooling, but with the project.json tooling we set all builds to 0.0.0 unless they were getting set from the outside for two reasons, means you know a build is just a local build and means you don't have to keep bumping the version in this file. Can we continue doing that?